cmd/kubeless/trigger/update.go

```diff
@@ -0,0 +1,132 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package trigger
+
+import (
+	kubelessApi "github.com/kubeless/kubeless/pkg/apis/kubeless/v1beta1"
+	"github.com/kubeless/kubeless/pkg/utils"
+	"github.com/robfig/cron"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var updateCmd = &cobra.Command{
+	Use:   "update <trigger_name> FLAG",
+	Short: "Update a trigger to Kubeless",
+	Long:  `Update a trigger to Kubeless`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) != 1 {
+			logrus.Fatal("Need exactly one argument - trigger name")
+		}
+		triggerName := args[0]
+
+		triggerHTTP, err := cmd.Flags().GetBool("trigger-http")
+		if err != nil {
+			logrus.Fatal(err)
+		}
+
+		schedule, err := cmd.Flags().GetString("schedule")
+		if err != nil {
+			logrus.Fatal(err)
+		}
+
+		if schedule != "" {
+			if _, err := cron.ParseStandard(schedule); err != nil {
+				logrus.Fatalf("Invalid value for --schedule. " + err.Error())
+			}
+		}
+
+		ns, err := cmd.Flags().GetString("namespace")
+		if err != nil {
+			logrus.Fatal(err)
+		}
+		if ns == "" {
+			ns = utils.GetDefaultNamespace()
+		}
+
+		topic, err := cmd.Flags().GetString("trigger-topic")
+		if err != nil {
+			logrus.Fatal(err)
+		}
+
+		functionName, err := cmd.Flags().GetString("function-name")
+		if err != nil {
+			logrus.Fatal(err)
+		}
+
+		kubelessClient, err := utils.GetKubelessClientOutCluster()
+		if err != nil {
+			logrus.Fatalf("Can not out-of-cluster client: %v", err)
+		}
+
+		_, err = utils.GetFunction(kubelessClient, functionName, ns)
+		if err != nil {
+			logrus.Fatalf("Unable to find the function %s in the namespace %s. Received %s: ", functionName, ns, err)
+		}
+
+		trigger := kubelessApi.Trigger{}
+		trigger.TypeMeta = metav1.TypeMeta{
+			Kind:       "Trigger",
+			APIVersion: "kubeless.io/v1beta1",
+		}
+
+		switch {
+		case triggerHTTP:
+			trigger.Spec.Type = "HTTP"
+			trigger.Spec.Topic = ""
+			trigger.Spec.Schedule = ""
+			break
+		case schedule != "":
+			trigger.Spec.Type = "Scheduled"
+			trigger.Spec.Schedule = schedule
+			trigger.Spec.Topic = ""
+			break
+		case topic != "":
+			trigger.Spec.Type = "PubSub"
+			trigger.Spec.Topic = topic
+			trigger.Spec.Schedule = ""
+			break
+		}
+
+		trigger.Spec.FunctionName = functionName
+		trigger.ObjectMeta = metav1.ObjectMeta{
+			Name:      triggerName,
+			Namespace: ns,
+			Labels: map[string]string{
+				"created-by": "kubeless",
+			},
+		}
```

Comment:
all the `trigger` object constructions can be placed in a common function so the code is not duplicated here and in the `deploy` command